### ISSUE TITLE
Empty result when selecting fields for multiple measurements

### ISSUE THREAD

**Issue by mathiashaessig** [influxdata/influxdb#7956]
# **Bug Report**

**Version:** 1.2.0

**Steps to reproduce**

```
create database test
use test
insert m1,t=1 val1=1
insert m2,t=1 val1=2
select time,val1,val2 from m1,m2,foo
```

**Expected result (produced by versions < 1.2.0)**

```
name: m1
time                val1 val2
----                ---- ----
1486483348180868700 1

name: m2
time                val1 val2
----                ---- ----
1486483367293321100 2
```

**Result with version 1.2.0:**
`<empty>`

However, the following queries return the correct result:

```
select time,val1,val2 from m1,m2

select * from m1,m2,foo
```

**Comment by jsternberg** [comment]
Are you using a regular expression in `FROM`? Might be the same as #7877.

**Comment by mathiashaessig** [comment]
No, I am not using regular expressions in FROM. Just the queries listed in the examples. The workaround for me is to use
`select *::field from m1,m2,foo`
instead of listing the fields explicitly in the select part.

**Comment by jsternberg** [comment]
So this happens when `foo` is included in 1.2.0 and goes away when `foo` is removed? That's the only difference I can see. Is `foo` a valid measurement? If it is, can I see the result of `show tag keys from foo`?

This definitely seems like a bug. I'm just trying to narrow down where it might be coming from.

**Comment by mathiashaessig** [comment]
The result is empty when all select fields are listed explicitly and the from part contains an invalid measurement.
When using select * or select *::field it works even if the from part contains an invalid measurement.
